### ISSUE TITLE
[FEAT][MOE] fused_topk: softmax/sigmoid scoring + top-k selection + renormalize

### ISSUE THREAD

**Issue by michaelwithu** [tile-ai/TileOPs#588]
## Parent Issue
Closes part of #404

## Overview
Implement `fused_topk` kernel for MoE routing. This is the first stage of the MoE cutlass pipeline.

## Inputs / Outputs

```
Input:
  gating_output  [T, E]   # router linear output (logits), bias correction done externally

Output:
  topk_weights   [T, K]   # selected expert weights (float32)
  topk_ids       [T, K]   # selected expert indices (int32)
```

## Kernel Behavior

For each token row (independent per row):
1. **Scoring**: apply softmax or sigmoid to the `E` logits (numerically stable)
2. **Top-K selection**: find the K highest-scoring experts via warp-level reduction
3. **Renormalize** (optional, controlled by caller): re-normalize selected K weights to sum to 1

## Design Decisions
- **bias correction**: done externally before passing `gating_output` in — not fused
- **scoring_func**: runtime parameter, supports `"softmax"` and `"sigmoid"`
- **renormalize**: runtime bool, caller decides

## Reference
- vLLM: `vllm/model_executor/layers/fused_moe/router/fused_topk_router.py`
- vLLM CUDA: `csrc/moe/topk_softmax_kernels.cu`

## Acceptance Criteria
- [ ] Correctness test against PyTorch reference (softmax path)
- [ ] Correctness test against PyTorch reference (sigmoid path)
- [ ] Correctness test with renormalize=True
- [ ] Benchmark vs vLLM baseline
